mofacts/client/views/experiment/card.js

```diff
@@ -2890,9 +2890,11 @@ async function processUserTimesLog() {
     // Initialize client side student performance
     const curUser = Meteor.user();
     const currentTdfId = Session.get('currentTdfId');
+    const curTdf = Session.get('currentTdfFile');
+    const curTdfUnit = curTdf.tdfs.tutor.unit[0];
     await setStudentPerformance(curUser._id, curUser.username, currentTdfId);
 
-    if (needFirstUnitInstructions) {
+    if (needFirstUnitInstructions && (curTdfUnit.unitinstructions != "" || typeof curTdfUnit.unitinstructions !== 'undefined')) {
```

Comment:
something cannot be equal to both "" and undefined. This logic will always return true.